### PR TITLE
Deprecate string.isEmptyString for string.isEmpty

### PR DESCRIPTION
**Pull request by ben-albrecht**
This PR deprecates `string.isEmptyString` for `string.isEmpty`, standardizing the interface for methods asking "is this thing empty?" in Chapel's standard library.

Closes https://github.com/chapel-lang/chapel/issues/11437

### CODE REVIEW

File: modules/internal/String.chpl [hunk 1]
```diff
@@ -807,11 +807,19 @@ module String {
       compilerError("substring removed: use string[range]");
     }
 
+    /*
+     Deprecated, use :proc:`string.isEmpty`.
+     */
+    inline proc isEmptyString() : bool {
+      compilerWarning("'isEmptyString' has been deprecated, use 'isEmpty' method instead");
```
Comment (ronawho) [inline modules/internal/String.chpl]: I'd go with something like `compilerWarning("isEmptyString is deprecated - please use isEmpty instead")` to be more consistent with other deprecation warnings (which also makes me think we should have a deprecatedWarning wrapper)

Comment (ben-albrecht) [reply modules/internal/String.chpl]: Agreed. Want to open an issue on that or shall I?

File: test/deprecated/string-isEmptyString.chpl [hunk 2]
```diff
@@ -0,0 +1 @@
+var b = 'deprecate me please'.isEmptyString();
```
Comment (ronawho) [inline test/deprecated/string-isEmptyString.chpl]: Kinda silly, but maybe test on an empty and non-empty string just to make sure the deprecated thing still works until it's removed?

File: modules/internal/String.chpl (hunk 1 above)
Comment (ronawho) [reply modules/internal/String.chpl]: Go for it

File: test/deprecated/string-isEmptyString.chpl (hunk 2 above)
Comment (ben-albrecht) [reply test/deprecated/string-isEmptyString.chpl]: sure

File: modules/internal/String.chpl (hunk 1 above)
Comment (ben-albrecht) [reply modules/internal/String.chpl]: Done: https://github.com/chapel-lang/chapel/issues/12438

**Comment by ben-albrecht** [comment]
I plan on merging this later this evening.

**Pull request merged**